### PR TITLE
Question fetchin and dispalying

### PR DESCRIPTION
**Pull request by Sandipan-Ghosh**
Updated files: questions-dashboard.html
redmine task number-20905, i.e, client side implementation to display the questions is completed.

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Core/Controllers/QuestionController.cs [hunk 1]
```diff
@@ -44,5 +44,11 @@ public IActionResult AddCodeSnippetQuestion([FromBody]CodeSnippetQuestionDto cod
 
             return Ok(codeSnippetQuestionDto);
         }
+        [HttpGet("question")]
+        public IActionResult GetAllQuestions()
+        {
+            var questionsList = _questionsRepository.GetAllQuestions();
+            return Json(questionsList);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/QuestionController.cs]: the return type is not proper.

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/QuestionFetching/QuestionFetchingDto.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Promact.Trappist.DomainModel.Models.Question;
+using System.Collections.Generic;
+
+namespace Promact.Trappist.DomainModel.ApplicationClasses.QuestionFetchingDto
+{
+    /// <summary>
+    /// Application class for SingleMultipleAnswerQuestion
+    /// </summary>
+    public class QuestionFetchingDto:QuestionBase
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/QuestionFetching/QuestionFetchingDto.cs]: please rename it to application class. remove dto specific conventions.

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 3]
```diff
@@ -20,10 +22,12 @@ public QuestionRepository(TrappistDbContext dbContext)
...
         /// </summary>
         /// <returns>Question list</returns>
-        public List<SingleMultipleAnswerQuestion> GetAllQuestions()
+        public ICollection<QuestionFetchingDto> GetAllQuestions()
         {
-            var question = _dbContext.SingleMultipleAnswerQuestion.ToList();
-            return question;
+            var questions = _dbContext.SingleMultipleAnswerQuestion.ProjectTo<QuestionFetchingDto>().ToList();
+            questions.AddRange(_dbContext.CodeSnippetQuestion.ProjectTo<QuestionFetchingDto>().ToList());
+            return (questions);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: what is the purpose of closing and opening brackets around variable?

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 4]
```diff
@@ -20,10 +22,12 @@ public QuestionRepository(TrappistDbContext dbContext)
         /// Get all questions
         /// </summary>
         /// <returns>Question list</returns>
-        public List<SingleMultipleAnswerQuestion> GetAllQuestions()
+        public ICollection<ApplicationClass> GetAllQuestions()
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: the name should be proper.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/category.model.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿export class Category
+{
+    categoryName: string;
+    id: number;
+    createdDateTime: Date;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/category.model.ts]: no need to take createdDateTime and updatedDateTime at client side. updateddatetime spelling is also wrong.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts [hunk 6]
```diff
@@ -1,4 +1,15 @@
...
+﻿import { DifficultyLevel } from "../questions/enum-difficultylevel"
+import { QuestionType } from "../questions/enum-questiontype"
+import { Option } from "../questions/option.model";
+import{Category} from "../questions/category.model"
+export class Question {
+    id: number;
+    questionDetail: string;
+    questionType: QuestionType;
+    difficultyLevel: DifficultyLevel;
+    createdBy: string;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/question.model.ts]: no need to take createdBy and updatedBy here.

**Comment by Sandipan-Ghosh** [comment]
Files changed are :
QuestionController.cs
SingleMultipleAnswerQuestionApplicationClass.cs
SingleMultipleAnswerQuestion.cs
CategoryRepository.cs
ICategoryRepository.cs
IQuestionRespository.cs
QuestionRepository.cs
Startup.cs
category.model.ts
enum-difficultylevel.ts
enum-questiontype.ts
option.model.ts
question.model.ts
questions-dashboard.component.ts
questions-dashboard.html
questions.module.ts
systemjs.config.js

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/enum-questiontype.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+﻿export enum QuestionType {
+    single = 0,
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/enum-questiontype.ts]: please make it proper like singleanswer, multipleanswer, codesnippet question.

File: Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js [hunk 8]
```diff
@@ -39,7 +39,7 @@
                 main: 'index.js',
                 defaultExtension: 'js'
             },
-            '.': { defaultExtension: 'js' }
+            '.': { defaultExtension: 'js' },
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js]: why have you added the comma at the end?

File: Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js [hunk 9]
```diff
@@ -25,7 +25,7 @@
             '@angular/material': 'npm:@angular/material/bundles/material.umd.js',
             // other libraries
             'rxjs': 'npm:rxjs',
-            'angular2-wizard': 'npm:angular2-wizard/dist/'
+            'angular2-wizard': 'npm:angular2-wizard/dist/',
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js]: why have you added the comma at the end?

File: Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/QuestionFetching/SingleMultipleAnswerQuestionApplicationClass.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
+﻿using Promact.Trappist.DomainModel.Models.Question;
+using System.Collections.Generic;
+
+namespace Promact.Trappist.DomainModel.ApplicationClasses.QuestionFetchingDto
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/ApplicationClasses/QuestionFetching/SingleMultipleAnswerQuestionApplicationClass.cs]: please remove Dto specific namespace. make it proper.

**Pull request merged**